### PR TITLE
Layered icons

### PR DESCRIPTION
**Pull request by rpiaggio**
Notably absent from `react-fontawesome` is the ability to create layered icons via components (you can always do it by manually adding classes to tags). See https://github.com/FortAwesome/react-fontawesome/issues/225 and https://github.com/FortAwesome/react-fontawesome/issues/419.

This PR adds this missing feature by providing a component API on top of fontawesome's core JS API (which is basically what `react-fontawesome` does).

To achieve this, I had to duplicate some of the logic from `react-fontawesome`. Namely: resolving classes for each property and converting the abstract tag defintion the core layer returns into React's virtual DOM.

I kept the sole component from `react-fontawesome`: `FontAwesomeIcon`. But I wonder if we should switch that to our own logic too (which would be pretty straightforward) and scrap `react-fontawesome` altogether.

I added a demo project which outputs a layered circle icon with the letter `H` and a counter showing the number `23`:

image

### CODE REVIEW

File: font-awesome/src/main/scala/lucuma/react/fa/interop.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+package lucuma.react.fa
+
+import cats.syntax.all.*
+import japgolly.scalajs.react.*
+import japgolly.scalajs.react.vdom.*
+
+import scalajs.js
+
+extension (ae: AbstractElement)
```
Comment (cquiroz) [inline font-awesome/src/main/scala/lucuma/react/fa/interop.scala:12]: I guess this is based on what the js library does?

**Comment by cquiroz** [comment]
Looks great, We could as you say switch completely off `react-fontawesome`?
I think we still need the facade to "install" the icons?

**Review by cquiroz — APPROVED** [review]
LGTM

File: font-awesome/src/main/scala/lucuma/react/fa/interop.scala (hunk 1 above)
Comment (rpiaggio) [reply font-awesome/src/main/scala/lucuma/react/fa/interop.scala:12]: Yes, the JS library returns an "abstract" representation of the tag + attributes + children and the react library turns that into Vdom. Here we do it with scalajs-react idioms.

**Comment by rpiaggio** [comment]
> Looks great, We could as you say switch completely off `react-fontawesome`? I think we still need the facade to "install" the icons?

Yes, the JS API still needs us to add the icons to the library in order to do proper tree shaking.

I'll attempt to bypass `react-fontawesome` completely.

**Review by toddburnside — APPROVED** [review]
Cool! Looks great

File: build.sbt [hunk 2]
```diff
@@ -384,9 +385,19 @@ lazy val fontAwesome = project
   .dependsOn(common)
   .settings(
     name := "lucuma-react-font-awesome",
+    Compile / scalacOptions += "-language:implicitConversions",
     facadeSettings
   )
 
+lazy val fontAwesomeDemo = project
```
Comment (armanbilge) [inline build.sbt:392]: We should add a `rootFontAwesome` that aggregate this demo project so that it gets compiled in CI.

**Pull request merged**